plugin/restore/mysql/mysql.go

```diff
@@ -357,22 +481,22 @@ func (r *Restore) SyncArchivedBinlogFiles(ctx context.Context, instance *api.Ins
 }
 
 // SyncLatestBinlog syncs the latest binlog between the instance and `binlogDir`
-func (r *Restore) SyncLatestBinlog(ctx context.Context, instance *api.Instance, binlogDir string) error {
+func (r *Restore) SyncLatestBinlog(ctx context.Context, binlogDir string) error {
 	latestBinlogFileOnServer, err := r.getLatestBinlogFileMeta(ctx)
 	if err != nil {
 		return err
 	}
-	return r.downloadBinlogFile(ctx, instance, binlogDir, *latestBinlogFileOnServer)
+	return r.downloadBinlogFile(ctx, binlogDir, *latestBinlogFileOnServer)
 }
 
 // downloadBinlogFile syncs the binlog specified by `meta` between the instance and local.
-func (r *Restore) downloadBinlogFile(ctx context.Context, instance *api.Instance, binlogDir string, binlog mysql.BinlogFile) error {
+func (r *Restore) downloadBinlogFile(ctx context.Context, binlogDir string, binlog mysql.BinlogFile) error {
 	// for mysqlbinlog binary, --result-file must end with '/'
 	resultFileDir := strings.TrimRight(binlogDir, "/") + "/"
 	// TODO(zp): support ssl?
 	cmd := exec.CommandContext(ctx, r.mysqlutil.GetPath(mysqlutil.MySQL),
 		binlog.Name,
-		fmt.Sprintf("--read-from-remote-server --host=%s --port=%s --user=%s --password=%s", instance.Host, instance.Port, instance.Username, instance.Password),
+		fmt.Sprintf("--read-from-remote-server --host=%s --port=%s --user=%s --password=%s", r.connCfg.Host, r.connCfg.Port, r.connCfg.Username, r.connCfg.Password),
 		"--raw",
 		fmt.Sprintf("--result-file=%s", resultFileDir),
 	)
```

Comment:
IMO we can make the state less later.
The /tmp solution is not perfect either, it's usually only cleared after a reboot, and we have less control over it.

Also, the `os.Remove` is safe here, because the binlog file is always synced from the server. If it's gone or corrupted, just re-download.
Moving to the /tmp and move back will definitely make the solution harder to get right: we need to handle all kinds of fs errors, and this could leave partial downloaded files everywhere.

Reply:
> Also I am curious whether whether we need to persist the binlog file in our data directory at all or can just be left in the /tmp.
Under normal circumstances, the frequency of users initiating PITR is not high, and saving or not saving will not bring much difference, although we cannot clean up mysqlbinlog yet.
But this problem is consistent with using runner to synchronize mysqlbinlog. If we do not keep mysqlbinlog, then there is no need for `runner` to exist.
We consider that users do not understand the relationship between mysqlbinlog and full backup. If we do not keep mysqlbinlog, once mysqlbinlog is deleted, users may not be able to use PITR.
The design here requires `runner`, `pitr`, and `full backup strategies` to cooperate.

Reply:
> Also I am curious whether whether we need to persist the binlog file in our data directory at all or can just be left in the /tmp.  

Under normal circumstances, the frequency of users initiating PITR is not high, and saving or not saving will not bring much difference, although we cannot clean up mysqlbinlog yet.
But this problem is consistent with using runner to synchronize mysqlbinlog. If we do not keep mysqlbinlog, then there is no need for runner to exist.
We consider that users do not understand the relationship between mysqlbinlog and full backup. If we do not keep mysqlbinlog, once mysqlbinlog is deleted, users may not be able to use PITR.
The design here requires runner, pitr, and full backup strategies to cooperate.